packages/devtools_app/lib/src/trees.dart

```diff
@@ -137,6 +137,18 @@ class TreeNode<T extends TreeNode<T>> {
       returnCondition: condition,
     );
   }
+
+  T firstNodeAtLevel(int level) {
+    if (level >= depth) return null;
+    var node = this;
+    while (node.level < level) {
+      // Walk down the tree until we find the node at [level].
+      if (node.children.isNotEmpty) {
+        node = node.children.firstWhere((n) => n.depth > level - n.level);
```

Comment:
this will throw an exception if no children match the level. Is that intended. It would be good to add a test for this helper.

Reply:
makes sense. Maybe add a comment to clarify.

Reply:
done

Reply:
Maybe this method needs a top level comment. There are some specific assumptions about what the depth and level